clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -1804,18 +1804,20 @@ ::mlir::Attribute ConstArrayAttr::parse(::mlir::AsmParser &parser,
 
   // ArrayAttrrs have per-element type, not the type of the array...
   if (resultVal->dyn_cast<ArrayAttr>()) {
-    // Parse literal ':'
-    if (parser.parseColon())
-      return {};
-
-    // Parse variable 'type'
-    resultTy = ::mlir::FieldParser<::mlir::Type>::parse(parser);
-    if (failed(resultTy)) {
-      parser.emitError(
-          parser.getCurrentLocation(),
-          "failed to parse ConstArrayAttr parameter 'type' which is "
-          "to be a `::mlir::Type`");
-      return {};
+    // Array has implicit type: infer from const array type.
+    if (parser.parseOptionalColon()) {
+      resultTy = type;
+    }
```

Comment:
put the else with the rbrace `} else { //`. Did `clang-format` formatted like this? I don't remember if I already asked you if you usually clang-format your patches, it's a good practice if you don't.

Reply:
I do use the VS Code clangd extension (which enables clang-format) for formatting, but it didn't seem to mind this comment. Do you use a custom `.clang-format` style file?

Reply:
I use the clang-format extension by Xaver Hellauer. I also have in my JSON pref:

```
"editor.defaultFormatter": "xaver.clang-format",
"editor.formatOnSaveMode": "modifications",
"editor.formatOnSave": true,
```

which automatically formats when I save, but only for lines I've modified (some extensions will format the whole file instead).